streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/execution/http/PipelineExecutor.java

```diff
@@ -107,7 +108,7 @@ private void updateGroupIds(InvocableStreamPipesEntity entity) {
             .filter(is -> is.getEventGrounding().getTransportProtocol() instanceof KafkaTransportProtocol)
             .map(is -> is.getEventGrounding().getTransportProtocol())
             .map(KafkaTransportProtocol.class::cast)
-            .forEach(tp -> tp.setGroupId(UUID.randomUUID().toString()));
+            .forEach(tp -> tp.setGroupId(pipeline.getName() + "-" + MD5.crypt(tp.getElementId())));
```

Comment:
Can we use pipeline.getId() instead of the name to avoid having blank spaces from the user-defined pipeline name in the group Id?

Reply:
The pipeline name is used instead of id, just to make it easier to identify in Kafka monitoring. Certainly if the UI interface does not check special characters, such as spaces, surely we should use id to generate group id.